src/eviction/eviction.c

```diff
@@ -67,40 +67,62 @@ static inline uint32_t ocf_evict_do(ocf_cache_t cache,
 	uint32_t to_evict = 0, evicted = 0;
 	struct ocf_user_part *part;
 	ocf_part_id_t part_id;
+	unsigned overflow_size;
+	int overflown;
 
-	/* For each partition from the lowest priority to highest one */
-	for_each_part(cache, part, part_id) {
+	/* start with overflown partitions */
+	for (overflown = 1; overflown >= 0; overflown--)
```

Comment:
Whoa, this looks a bit unusual. How about moving all this stuff to separate function taking `overflown` (or even better `overflown_only`) as an argument?

Reply:
done